transcom/mymove
# INT I-13434 & B-21432 updating market code for all shipment types

## [B-21432](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21432)
## [I-13434](https://www13.v1host.com/USTRANSCOM38/Issue.mvc/Summary?oidToken=Issue%3A1024847)

## Summary

An issue was found where NTS & NTSR shipment market codes weren't being updated on update. I took this opportunity to move the market code logic to the shipments model as well as add an additional helper function for PPM shipments that use a different model type.

All shipments should now be included in the model function and I addressed creation/update of all shipment types.

### How to test

1. You'll need to turn on the `FEATURE_FLAG_ENABLE_ALASKA` and `FEATURE_FLAG_UNACCOMPANIED_BAGGAGE` flags in `.envrc`
2. Fire up your `server` and `client`
3. Log in as a customer and create a move
4. Create a shipment of each type using an alaska address
5. Confirm that the market code is `i` next to the shipment
6. Now go through each shipment and update the state to NOT be Alaska (all you gotta update is the state)
7. Confirm the market code changes to `d` next to the shipment
8. Repeat this same process as an office user creating shipments
9. To earn mega bonus points, reverse the creation/update process by creating a domestic shipment and then updating each shipment type to use an AK address that should change it from `d` -> `i`
